### PR TITLE
Add "Go v1.17 or more" as prerequisite for contributers and upgrade golangci-lint to v1.46.2 for that

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Add "Go v1.17 or more" as prerequisite to CONTRIBUTING guide.
The difference of Go versions sometimes causes some issues. One of them is the result of generating file. (`make openapi` in our case)

And we need to update golangci-lint to use it with go v1.18. So, this branch upgrade that to latest version.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: simulator/.golangci.yml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 linters-settings:
+  # if you fix this gci config, you should modify the gci command on `make format` as well.
   gci:
-    local-prefixes: sigs.k8s.io/kube-scheduler-simulator/simulator
+    sections:
```
Comment (sanposhiho) [inline simulator/.golangci.yml:4]: The previous configuration `local-prefixes` is deprecated.

File: simulator/Makefile [hunk 2]
```diff
@@ -13,6 +13,7 @@ lint:
 
 .PHONY: format
 format:
+	find . -name "*.go" | grep -v "vendor" | grep -v "submodule" | xargs gci write --Section Standard --Section Default --Section "Prefix(sigs.k8s.io/kube-scheduler-simulator/simulator)"
```
Comment (sanposhiho) [inline simulator/Makefile:16]: gci's autofix feature is disabled in this current golangci-lint version. So, manually run it on format.

File: simulator/Makefile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 .PHONY: generate
-generate: vendor
-	go generate ./... && \
+generate: format
```
Comment (sanposhiho) [inline simulator/Makefile:2]: now the mock files are checked by gci. So, mocks need to be formated after generated.

**Comment by 196Ikuchil** [comment]
Nice work!
It worked fine locally, too!

/lgtm

**Pull request merged**